src/testing/testing.go

```diff
@@ -1034,6 +1034,52 @@ func (c *common) logDepth(s string, depth int) {
 	}
 }
 
+func (c *common) Output() io.Writer {
+	return OutputWriter{c}
+}
+
+type OutputWriter struct {
+	c *common
+}
+
+func (s OutputWriter) Write(p []byte) (int, error) {
+	// Temporary hack to get the proof of concept working.
+	str := string(p)
+	depth := 3
+
+	// Code taken almost unchanged from c.logDepth().
+	// The return values are set to default as a quick fix.
+	s.c.mu.Lock()
+	defer s.c.mu.Unlock()
+	if s.c.done {
+		// This test has already finished. Try and log this message
+		// with our parent. If we don't have a parent, panic.
+		for parent := s.c.parent; parent != nil; parent = parent.parent {
+			parent.mu.Lock()
+			defer parent.mu.Unlock()
+			if !parent.done {
+				parent.output = append(parent.output, parent.decorate(str, depth+1)...)
```

Comment:
This `Write` method is at a lower level than logDepth. It isn't responsible for adding line info or a final newline, so it shouldn't call `decorate`.  It does need to handle indentation, though. I'm not sure where that happens.